[Elixir] New concept exercise: access (#1575)
exercism/v3: languages/elixir/exercises/concept/access/.docs/introduction.md

> Elixir uses code **Behaviors** to provide a common generic interfaces while facilitating specific implementations for each module which implements it. One such common example is the **Access Behavior**.

Author: ```suggestion
Elixir uses code _Behaviours_ to provide a common generic interfaces while facilitating specific implementations for each module which implements it. One such common example is the _Access Behaviour_.
```

## Access Behaviour

> The _Access Behaviour_ provides a common interface for retrieving key-based data from a data structure: like from a key-value map structure. Elixir supports the _Access Behaviour_ for maps and keyword lists, but let's look at its use for maps to get a feel for it. _Access Behavior_ specifies that when you have a map, you may follow it with _square brackets_ and then use the key to retrieve the value associated with that key.

Reviewer: I'm not sure which form you intended to use, but "Access Behavior" is mixed up with "Access Behaviour", and I guess we should stick to just one.

> The _Access Behavior_ provides a common interface for retrieving key-based data from a data structure: like from a key-value map structure. Elixir supports the _Access Behavior_ for maps and keyword lists, but let's look at its use for maps to get a feel for it. _Access Behavior_ specifies that when you have a map, you may follow it with _square brackets_ and then use the key to retrieve the value associated with that key.

Author: ```suggestion
The _Access Behaviour_ provides a common interface for retrieving key-based data from a data structure: like from a map data structure. Elixir supports the _Access Behaviour_ for maps and keyword lists, but let's look at its use for maps to get a feel for it. _Access Behaviour_ specifies that when you have a map, you may follow it with _square brackets_ and then use the key to retrieve the value associated with that key.
```

```elixir
# Suppose we have these two maps defined (note the difference in the key type)
my_map = %{key: "my value"}
your_map = %{"key" => "your value"}

# Obtain the value using the Access Behavior
my_map[:key] == "my value"

> your_map["key"] == nil

Author: ```suggestion
your_map[:key] == nil
```
